bin/ch-grow.py.in

```diff
@@ -1,169 +1,44 @@
-                       "help": argparse.SUPPRESS }
-         p.add_argument(*args, **kwargs)
-   add_opts(ap, common_opts, False)
-
-   # build
-   help="build image from Dockerfile"
-   sp = sps.add_parser("build", help=help, description=help,
-                       formatter_class=ch.HelpFormatter)
-   sp.set_defaults(func=build.main)
-   add_opts(sp, common_opts, True)
-   sp.add_argument("-b", "--bind", metavar="SRC[:DST]",
-                   action="append", default=[],
-                   help="mount SRC at guest DST (default /mnt/0, /mnt/1, etc.)")
-   sp.add_argument("--build-arg", metavar="ARG[=VAL]",
-                   action="append", default=[],
-                   help="set build-time variable ARG to VAL, or $ARG if no VAL")
-   sp.add_argument("-f", "--file", metavar="DOCKERFILE",
-                   help="Dockerfile to use (default: CONTEXT/Dockerfile)")
-   sp.add_argument("-n", "--dry-run", action="store_true",
-                   help="don't execute instructions")
-   sp.add_argument("--no-fakeroot", action="store_true",
-                   help="don't try unprivileged build workarounds")
-   sp.add_argument("--parse-only", action="store_true",
-                   help="stop after parsing the Dockerfile")
-   sp.add_argument("-t", "--tag", metavar="TAG",
-                   help="name of image to create (default: inferred)")
-   sp.add_argument("context", metavar="CONTEXT",
-                   help="context directory")
-
-   # list
-   help="list images in storage"
-   sp = sps.add_parser("list", help=help, description=help,
-                       formatter_class=ch.HelpFormatter)
-   sp.set_defaults(func=misc.list_)
-   add_opts(sp, common_opts, True)
-
-   # pull
-   help="pull image from remote repository to local filesystem"
-   sp = sps.add_parser("pull", help=help, description=help,
-                       formatter_class=ch.HelpFormatter)
-   sp.set_defaults(func=misc.pull)
-   add_opts(sp, common_opts, True)
-   sp.add_argument("--last-layer", metavar="N", type=int,
-                   help="stop after unpacking N layers")
-   sp.add_argument("--parse-only", action="store_true",
-                   help="stop after parsing the image reference")
-   sp.add_argument("image_ref", metavar="IMAGE_REF", help="image reference")
-   sp.add_argument("image_dir", metavar="IMAGE_DIR", nargs="?",
-                   help="unpacked image path (default: opaque path in storage dir)")
-
-   # storage-path
-   help="print storage directory path"
-   sp = sps.add_parser("storage-path", help=help, description=help,
-                       formatter_class=ch.HelpFormatter)
-   sp.set_defaults(func=misc.storage_path)
-   add_opts(sp, common_opts, True)
-
-   # Parse it up!
-   if (len(sys.argv) < 2):
-       ap.print_help(file=sys.stderr)
-       sys.exit(1)
-   cli = ap.parse_args()
-   ch.log_setup(cli.verbose)
-
-   # Dispatch.
-   cli.func(cli)
-
-
-## Bootstrap ##
-
-if (__name__ == "__main__"):
-   main()
+#!/bin/bash
```

Comment:
I think this script can be simplified? Does it work to pass all arguments to `ch-image` and let it handle `--help`, `--version`, etc.? This can be done with `/bin/sh`.

Reply:
If we print our warning and then pass `--help` to `ch-image`, the regex `^(?:[Uu]sage:[^\n]+\n| +[^\n]+\n|\n)*([^\n]+)\n` in `docs-sane` fails. The warning would have to occur after passing the arguments and that would undermine the goal of having a warning followed by a `sleep(5)`. Alternatively we could update the regex but that seems awkward given this is a temporary script.